<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SerialPort</class>
 <widget class="QMainWindow" name="SerialPort">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>818</width>
    <height>725</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SerialPort</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_16">
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout_8">
      <item row="0" column="0">
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="0" column="0">
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="0" column="0">
           <widget class="QGroupBox" name="groupBox_3">
            <property name="font">
             <font>
              <family>微软雅黑 Light</family>
              <pointsize>10</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="title">
             <string>串口设置</string>
            </property>
            <layout class="QGridLayout" name="gridLayout">
             <item row="0" column="0">
              <widget class="QLabel" name="label_port">
               <property name="font">
                <font>
                 <family>微软雅黑 Light</family>
                 <pointsize>10</pointsize>
                 <weight>75</weight>
                 <bold>true</bold>
                </font>
               </property>
               <property name="text">
                <string>串口号</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QComboBox" name="CB_port"/>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_baudrate">
               <property name="font">
                <font>
                 <family>微软雅黑 Light</family>
                 <pointsize>10</pointsize>
                 <weight>75</weight>
                 <bold>true</bold>
                </font>
               </property>
               <property name="text">
                <string>波特率</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QComboBox" name="CB_baudrate">
               <property name="currentText">
                <string>9600</string>
               </property>
               <item>
                <property name="text">
                 <string>9600</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>38400</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>56700</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>115200</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>119600</string>
                </property>
               </item>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="label_databit">
               <property name="font">
                <font>
                 <family>微软雅黑 Light</family>
                 <pointsize>10</pointsize>
                 <weight>75</weight>
                 <bold>true</bold>
                </font>
               </property>
               <property name="text">
                <string>数据位</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QComboBox" name="CB_databit">
               <item>
                <property name="text">
                 <string>5</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>6</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>7</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>8</string>
                </property>
               </item>
              </widget>
             </item>
             <item row="3" column="0">
              <widget class="QLabel" name="label_checkbit">
               <property name="font">
                <font>
                 <family>微软雅黑 Light</family>
                 <pointsize>10</pointsize>
                 <weight>75</weight>
                 <bold>true</bold>
                </font>
               </property>
               <property name="text">
                <string>校验位</string>
               </property>
              </widget>
             </item>
             <item row="3" column="1">
              <widget class="QComboBox" name="CB_checkbit">
               <item>
                <property name="text">
                 <string>None</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>奇校验</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>偶校验</string>
                </property>
               </item>
              </widget>
             </item>
             <item row="4" column="0">
              <widget class="QLabel" name="label_stopbit">
               <property name="font">
                <font>
                 <family>微软雅黑 Light</family>
                 <pointsize>10</pointsize>
                 <weight>75</weight>
                 <bold>true</bold>
                </font>
               </property>
               <property name="text">
                <string>停止位</string>
               </property>
              </widget>
             </item>
             <item row="4" column="1">
              <widget class="QComboBox" name="CB_stopbit">
               <item>
                <property name="text">
                 <string>1</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>2</string>
                </property>
               </item>
              </widget>
             </item>
             <item row="5" column="0">
              <widget class="QLabel" name="label_open">
               <property name="font">
                <font>
                 <family>微软雅黑 Light</family>
                 <pointsize>10</pointsize>
                 <weight>75</weight>
                 <bold>true</bold>
                </font>
               </property>
               <property name="text">
                <string>串口操作</string>
               </property>
              </widget>
             </item>
             <item row="5" column="1">
              <widget class="QPushButton" name="Btn_open">
               <property name="font">
                <font>
                 <family>微软雅黑 Light</family>
                 <pointsize>10</pointsize>
                 <weight>75</weight>
                 <bold>true</bold>
                </font>
               </property>
               <property name="text">
                <string>打开</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="1" column="0">
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="0" column="0">
             <widget class="QPushButton" name="Btn_save">
              <property name="font">
               <font>
                <family>微软雅黑 Light</family>
                <pointsize>10</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>保存窗口</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QPushButton" name="Btn_clearrx">
              <property name="font">
               <font>
                <family>微软雅黑 Light</family>
                <pointsize>10</pointsize>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>清除接收</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item row="1" column="0">
         <widget class="QGroupBox" name="groupBox">
          <property name="font">
           <font>
            <family>微软雅黑 Light</family>
            <pointsize>10</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="title">
           <string>接收设置</string>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
          <property name="checkable">
           <bool>false</bool>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="3" column="0">
            <widget class="QCheckBox" name="checkBox_rx_dispsend">
             <property name="text">
              <string>显示发送</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QCheckBox" name="checkBox_rx_hexdisp">
             <property name="text">
              <string>16进制显示</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QCheckBox" name="checkBox_rx_autoline">
             <property name="text">
              <string>自动换行</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QCheckBox" name="checkBox_rx_disptime">
             <property name="text">
              <string>显示时间</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="1">
       <widget class="QGroupBox" name="groupBox_4">
        <property name="font">
         <font>
          <family>微软雅黑</family>
          <pointsize>10</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="title">
         <string>串口接收数据</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="0">
          <widget class="QTextEdit" name="textEdit_rx"/>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <layout class="QGridLayout" name="gridLayout_15">
      <item row="0" column="0">
       <widget class="QGroupBox" name="groupBox_2">
        <property name="minimumSize">
         <size>
          <width>195</width>
          <height>180</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>195</width>
          <height>180</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>微软雅黑 Light</family>
          <pointsize>10</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="title">
         <string>发送设置</string>
        </property>
        <property name="flat">
         <bool>false</bool>
        </property>
        <property name="checkable">
         <bool>false</bool>
        </property>
        <layout class="QGridLayout" name="gridLayout_14">
         <item row="0" column="0">
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0">
            <widget class="QCheckBox" name="checkBox_tx_newline">
             <property name="text">
              <string>发送新行</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QCheckBox" name="checkBox_tx_hexsend">
             <property name="text">
              <string>16进制发送</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QCheckBox" name="checkBox_tx_delayed">
             <property name="text">
              <string>定时发送</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QWidget" name="widget" native="true">
             <property name="minimumSize">
              <size>
               <width>170</width>
               <height>35</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>170</width>
               <height>35</height>
              </size>
             </property>
             <layout class="QGridLayout" name="gridLayout_11">
              <property name="sizeConstraint">
               <enum>QLayout::SetFixedSize</enum>
              </property>
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>7</number>
              </property>
              <property name="rightMargin">
               <number>11</number>
              </property>
              <item row="0" column="0">
               <widget class="QLabel" name="label">
                <property name="minimumSize">
                 <size>
                  <width>90</width>
                  <height>23</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>90</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>周期(ms)：</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLineEdit" name="lineEdit_period">
                <property name="minimumSize">
                 <size>
                  <width>75</width>
                  <height>20</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>75</width>
                  <height>20</height>
                 </size>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QGridLayout" name="gridLayout_10">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox_5">
          <property name="font">
           <font>
            <family>微软雅黑</family>
            <pointsize>10</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="title">
           <string/>
          </property>
          <layout class="QGridLayout" name="gridLayout_9">
           <item row="0" column="0">
            <widget class="QTextEdit" name="textEdit_tx">
             <property name="sizeAdjustPolicy">
              <enum>QAbstractScrollArea::AdjustToContentsOnFirstShow</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="0" colspan="2">
         <layout class="QGridLayout" name="gridLayout_12">
          <item row="0" column="0">
           <widget class="QLineEdit" name="lineEdit_file"/>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="Btn_openfile">
            <property name="font">
             <font>
              <family>微软雅黑 Light</family>
              <pointsize>10</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>打开文件</string>
            </property>
            <property name="autoDefault">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QPushButton" name="Btn_sendfile">
            <property name="font">
             <font>
              <family>微软雅黑 Light</family>
              <pointsize>10</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>发送文件</string>
            </property>
            <property name="autoDefault">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QPushButton" name="Btn_stopsend">
            <property name="font">
             <font>
              <family>微软雅黑 Light</family>
              <pointsize>10</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>停止发送</string>
            </property>
            <property name="autoDefault">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="1">
         <layout class="QGridLayout" name="gridLayout_13">
          <item row="0" column="0">
           <widget class="QPushButton" name="Btn_send">
            <property name="font">
             <font>
              <family>微软雅黑 Light</family>
              <pointsize>10</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="cursor">
             <cursorShape>UpArrowCursor</cursorShape>
            </property>
            <property name="text">
             <string>发送</string>
            </property>
            <property name="autoRepeatDelay">
             <number>297</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="Btn_cleartx">
            <property name="font">
             <font>
              <family>微软雅黑 Light</family>
              <pointsize>10</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>清除发送</string>
            </property>
            <property name="autoDefault">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>layoutWidget</zorder>
   <zorder>groupBox_4</zorder>
   <zorder>groupBox_5</zorder>
   <zorder>lineEdit_period</zorder>
   <zorder>groupBox_2</zorder>
   <zorder></zorder>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionSend">
   <property name="text">
    <string>S:</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
